### PR TITLE
[openMFP] Move luigiContext and isInIframe to own composables

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
Move luigiContext and isInIframe to own composables

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: frontend/src/composables/useOpenMFP.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  }
+
+  return {
+    accountId,
+    luigiContext,
+    setRoute,
+    getLuigiContext,
+  }
+})
```
Comment (holgerkoser) [inline frontend/src/composables/useOpenMFP.js:82]:
```suggestion
import { useRoute } from 'vue-router'
import { until } from '@vueuse/core'
import { createGlobalState } from '@vueuse/core'
import LuigiClient from '@luigi-project/client'

import { useLogger } from '@/composables/useLogger'
import { useIsInIframe } from '@/composables/useIsInIframe'

export const useOpenMFP = createGlobalState((options = {}) => {
  const {
    logger = useLogger(),
    isInIframe = useIsInIframe(),
    route = useRoute(),
  } = options

  const luigiContext = ref(null)

  if (isInIframe.value) {
    logger.debug('Registering listener for Luigi context initialization and context updates')
    LuigiClient.addInitListener(context => setLuigiContext(context))
    LuigiClient.addContextUpdateListener(context => setLuigiContext(context))
    const pathname = toRef(route, 'path')
    watch(pathname, value => {
      if (value) {
        LuigiClient.linkManager().fromVirtualTreeRoot().withoutSync().navigate(value)
      }
    }, {
      immediate: true,
    })
  }

  function setLuigiContext (value) {
    luigiContext.value = value
  }

  const accountId = computed(() => luigiContext.value?.accountId)

  async function getLuigiContext () {
    if (!isInIframe.value) {
      return null
    }
    if (luigiContext.value !== null) {
      return luigiContext.value
    }
    const timeout = 3000
    try {
      await until(luigiContext).toBeTruthy({ 
        timeout: 1000, 
        throwOnTimeout: true,
      })
      return luigiContext.value
    } catch (err) {
      logger.error('The initialization of the Luigi Client has timed out after %d milliseconds', timeout)
      return null
    }
  }

  return {
    accountId,
    luigiContext,
    getLuigiContext,
  }
})
```

File: frontend/src/App.vue [hunk 2]
```diff
@@ -45,10 +46,10 @@ const configStore = useConfigStore()
 const loginStore = useLoginStore()
 const shootStore = useShootStore()
 const projectStore = useProjectStore()
-const appStore = useAppStore()
 const logger = inject('logger')
+const openMFP = useOpenMFP()
 
-appStore.setRoute(route)
+openMFP.setRoute(route)
```
Comment (holgerkoser) [inline frontend/src/App.vue:52]: move into composable (see below)

File: frontend/src/router/guards.js [hunk 3]
```diff
@@ -18,12 +18,14 @@ import { useSeedStore } from '@/store/seed'
...
 import { useTerminalStore } from '@/store/terminal'
 
+import { useOpenMFP } from '@/composables/useOpenMFP'
 import { useLogger } from '@/composables/useLogger'
 import { useApi } from '@/composables/useApi'
 
 export function createGlobalBeforeGuards () {
   const logger = useLogger()
   const api = useApi()
+  const openMFP = useOpenMFP()
```
Comment (holgerkoser) [inline frontend/src/router/guards.js:28]:
```suggestion
```

File: frontend/src/router/guards.js [hunk 4]
```diff
@@ -50,7 +52,7 @@ export function createGlobalBeforeGuards () {
         return true
       }
 
-      const context = await appStore.getLuigiContext()
+      const context = await openMFP.getLuigiContext()
```
Comment (holgerkoser) [inline frontend/src/router/guards.js:55]:
```suggestion
      const openMFP = useOpenMFP()
      const context = await openMFP.getLuigiContext()
```

File: frontend/src/App.vue [hunk 5]
```diff
@@ -31,10 +31,11 @@ import { useLoginStore } from '@/store/login'
 import { useLocalStorageStore } from '@/store/localStorage'
 import { useShootStore } from '@/store/shoot'
 import { useProjectStore } from '@/store/project'
-import { useAppStore } from '@/store/app'
 
 import { useCustomColors } from '@/composables/useCustomColors'
 
+import { useOpenMFP } from './composables/useOpenMFP'
```
[the remark this answers is not in the document]
Comment (petersutter) [reply frontend/src/App.vue:37]: ✅

File: frontend/src/composables/useOpenMFP.js (hunk 1 above)
Comment (petersutter) [reply frontend/src/composables/useOpenMFP.js:82]: ✅

File: frontend/src/router/guards.js (hunk 3 above)
Comment (petersutter) [reply frontend/src/router/guards.js:28]: ✅

File: frontend/src/router/guards.js (hunk 4 above)
Comment (petersutter) [reply frontend/src/router/guards.js:55]: ✅

File: frontend/src/App.vue (hunk 2 above)
Comment (petersutter) [reply frontend/src/App.vue:52]: ✅

File: frontend/src/composables/useOpenMFP.js [hunk 6]
```diff
@@ -34,49 +49,29 @@ export const useOpenMFP = createGlobalState(() => {
...
-      watch(pathname, value => {
-        if (value) {
-          LuigiClient.linkManager().fromVirtualTreeRoot().withoutSync().navigate(value)
-        }
-      }, {
-        immediate: true,
+    const timeout = 3000
+    try {
+      await until(luigiContext).toBeTruthy({
+        timeout: 1000,
```
Comment (holgerkoser) [inline frontend/src/composables/useOpenMFP.js:62]:
```suggestion
        timeout,
```

Comment (petersutter) [reply frontend/src/composables/useOpenMFP.js]: ✅

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Pull request merged**